dev-util/dput-ng: 3 new packages

nE0sIghT opened the pull request:
This PR provides new dev-util/dput-ng package with it's missing dependencies: dev-util/distro-info-data and dev-util/distro-info.

dput-ng tool is used for uploading Debian packages.

rindeal commented on dev-util/distro-info/distro-info-0.14.ebuild:
```diff
@@ -0,0 +1,66 @@
+
+	sed -i "/cd python && python/d" "${S}"/Makefile || die
+	sed -i "/\$(VENDOR)/d" "${S}"/Makefile || die
+	sed -i "/VENDOR ?=/d" "${S}"/Makefile || die
```
could use `-e` option instead of such duplication

rindeal commented on dev-util/dput-ng/dput-ng-1.10.ebuild:
```diff
@@ -0,0 +1,92 @@
+	interfaces
+	schemas
+	uploaders
+)
```
why do you introduce global arrays if they're only used in src_install`?

rindeal commented on dev-util/dput-ng/dput-ng-1.10.ebuild:
```diff
@@ -0,0 +1,92 @@
+		a2x --doctype manpage \
+			--format manpage \
+			-D "${S}"/man \
+			"${file}"
```
missing `die` + arrays are always better than multiline strings

nE0sIghT replied:
> arrays are always better than multiline strings

Could you please explain a bit more about this? Did you mean that it is better to define all man files as array?

rindeal commented:
distro-info Makefile needs patching of *FLAGS, it shouldn't call dpkg-* and should only use `+=` instead of `=`

idella commented:
ditto dev-util/devscripts

nE0sIghT commented:
Updated per comments in dev-util/devscripts PR

idella commented:
looking ok, shall get into the branch later to look more closely.

idella commented:
Again there are 2 instances of "${S}" set in src_compile() {, a minor issue.
`mkdir "${S}"/man  -> mkdir ./man || die`   This was mentioned before actually.

Setting DISTUTILS_SINGLE_IMPL=1 sets  python-single-r1 as the backend. This set the style of the whole ebuild in the python content.

Considering the number of suggestions made already, this looks fine now but for the missed "|| die"
Unlike the dev-util/devscripts, the state of the files on install all appear fine.  Should be ready on adding  the missed "|| die"

nE0sIghT commented:
Pushed new commits:

distro-info-data: disabled test phase because package provides only csv data and test script requires python.

distro-info:
1. Added missing perl dependency
2. Fixed installation path of perl library
3. Added test use flag and test dependency, test phase (all runs fine for me).
4. Removed ${S} from configure/compile/install

dput-ng:
1. added missing die
2. removed redundant ${S}

idella commented:
1. dev-util/dput-ng has a testsuite run by nosetests.  Found in dput-ng-1.10/work/dputng/tests

Since the testsuite might never have been added at all, one could overlook it.
On running it however, it reports FAILED (errors=2, failures=1) which represents 50%.

Please file this to upstream. To add this, you set dev-python/nose behind test use flag, then

```ebuild
python_test() {
        nosetests || die "Tests failed under ${EPYTHON}"
}
```
Being of debian origin I am not sure what to expect on probing the HOMEPAGE and seeking a tracker for the source repo.

2. dev-util/distro-info. Running the testsuite with python setup.py test in the source, it passes.
Running it as is from the ebuild, it returns an error value.

```
 ok 28 - Ubuntu unsupported
 * ERROR: dev-util/distro-info-0.14::gentoo failed (test phase):
```
However I think it is a flase negative. Otherwise the ebuild builds and installs

3. distro-info-data. The line `src_test() { :; }` achieves very little. If the data is incompatible with the file provided on which to test, fair enough.  More the norm to set RESTRICT="test" than this.

Also, for each package's commit msg, it is better to state that you are to maintain the package under Proxy Maintainers Project, and state the initial version of the package.

nE0sIghT commented:
distro-info-data: replaced src_test stub with RESTRICT="test"
distro-info: added setuptools and pylint to dependencies. Minor changes to src_test
dput-ng: added RESTRICT="test" since some tests are failing ([upstream bug submitted](https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=824652)). Added test dependencies and python_test function

Reworded commit messages.

idella commented:
I had missed emerging a dep to run testsuite of distro-info

All look fine now.

In dput-ng-1.10.ebuild;
I would shift the commentary lines 19, 20 to python_test() {
the line 21 to just following DEPEND.

idella commented:
One missed required entry
dev-python/python-debian requires [${PYTHON_USEDEP}]
otherwise I was ready to commit the 3 patches. Having just one out of order holds up the other two.
